### PR TITLE
Automatically fix old configs by default

### LINKED ISSUE #1257: Support old config versions automatically besides `skaffold fix`
Currently old versions can be upgraded with `skaffold fix`. Besides 'fixing' Skaffold should support older config versions automatically and just print out a warning message.

### PR DESCRIPTION
**Pull request by nkubala**
This adds a hook in the runner to automatically try and upgrade older configs to the latest version before running skaffold. This means users with older configs will no longer be required to run `skaffold fix` to run skaffold as we upgrade the config.

I'll leave `skaffold fix` in, in case anybody wants to use it to upgrade their config and overwrite it in their repo.

Fixes #1257

### CODE REVIEW

File: pkg/skaffold/schema/versions.go [hunk 1]
```diff
@@ -90,28 +91,24 @@ func ParseConfig(filename string, applyDefaults bool) (util.VersionedConfig, err
...
+	// first, check to make sure config version isn't too new
+	version, err := apiversion.Parse(vc.GetVersion())
 	if err != nil {
-		return errors.Wrap(err, "parsing api version")
+		return nil, errors.Wrap(err, "parsing api version")
 	}
-
-	if parsedVersion.LT(apiversion.MustParse(latest.Version)) {
-		return errors.New("config version out of date: run `skaffold fix`")
+	latestVersion := apiversion.MustParse(latest.Version)
```
Comment (r2d4) [inline pkg/skaffold/schema/versions.go]: you can move this up to the package level

File: pkg/skaffold/schema/versions.go [hunk 2]
```diff
@@ -90,28 +91,24 @@ func ParseConfig(filename string, applyDefaults bool) (util.VersionedConfig, err
...
+	if version.GT(latestVersion) {
+		return nil, errors.New("config version is too new for this version of skaffold: upgrade skaffold")
 	}
 
-	if parsedVersion.GT(apiversion.MustParse(latest.Version)) {
-		return errors.New("config version is too new for this version of skaffold: upgrade skaffold")
+	if version.LT(latestVersion) {
+		logrus.Infof("config version %s out of date: upgrading to latest (%s)", vc.GetVersion(), latest.Version)
```
Comment (r2d4) [inline pkg/skaffold/schema/versions.go]: Maybe we should print something out to stdout instead of just info to let the user know

Comment (r2d4) [reply pkg/skaffold/schema/versions.go]: Its not in the diff but how about something slightly more idiomatic

```
if version == latest, return latest
if version GT latest, return error

upgrade
return upgraded
```

File: pkg/skaffold/schema/versions.go [hunk 3]
```diff
@@ -90,27 +93,23 @@ func ParseConfig(filename string, applyDefaults bool) (util.VersionedConfig, err
...
+		return nil, fmt.Errorf("config version %s is too new for this version of skaffold: upgrade skaffold", vc.GetVersion())
 	}
 
-	return nil
-}
-
-// UpgradeToLatest upgrades a configuration to the latest version.
-func UpgradeToLatest(vc util.VersionedConfig) (util.VersionedConfig, error) {
-	var err error
+	out.Write([]byte(fmt.Sprintf("config version %s out of date: upgrading to latest (%s)\n", vc.GetVersion(), latest.Version)))
```
Comment (balopat) [inline pkg/skaffold/schema/versions.go]:
```suggestion
	out.Write([]byte(fmt.Sprintf("Warning! Config version %s out of date: converting to latest (%s). To permanently upgrade the config, run 'skaffold fix'.\n", vc.GetVersion(), latest.Version)))
```

**Pull request merged**